mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingLinearAlgorithm.scala

```diff
@@ -83,18 +83,20 @@ abstract class StreamingLinearAlgorithm[
       throw new IllegalArgumentException("Model must be initialized before starting training.")
     }
     data.foreachRDD { (rdd, time) =>
-      val initialWeights =
-        model match {
-          case Some(m) =>
-            m.weights
+      if (rdd.toLocalIterator.nonEmpty) {
+        val initialWeights =
```

Comment:
Ah, as it happens, I meant to follow up on one change we should have made yesterday in SPARK-8140 here: you can delete the entire `initialWeights` val here and replace it in the following line with `m.weights`. It will remove a warning. Can you pick that up too?